client/resp.go

```diff
@@ -53,14 +53,13 @@ func (c *Conn) handleOKPacket(data []byte) (*Result, error) {
 
 		//todo:strict_mode, check warnings as error
 		r.Warnings = binary.LittleEndian.Uint16(data[pos:])
-		pos += 2
 	} else if c.capability&CLIENT_TRANSACTIONS > 0 {
 		r.Status = binary.LittleEndian.Uint16(data[pos:])
 		c.status = r.Status
-		pos += 2
 	}
 
 	//new ok package will check CLIENT_SESSION_TRACK too, but I don't support it now.
+	// pos += 2
```

Comment:
It seems to me that it is still worth leaving the change of position next to the reading, and not taking it outside. Any future refactoring or code change can either make the code harder to understand or break the code.

Reply:
I've placed them back and commented them, I think that is the most readable way